### PR TITLE
chore(vim-patch.sh): replace hub with gh

### PR DESCRIPTION
**Pull request by zeertzjq**

### CODE REVIEW

**Comment by teto** [comment]
can't we fully replace hub with gh ?

**Comment by zeertzjq** [comment]
Can we assume everyone prefers `gh` to `hub`?

**Comment by teto** [comment]
rather than a matter of personal taste gh is the "official" tool https://github.com/cli/cli/issues/312 so if we can use one tool, instead of two let's go with gh whose future looks more promising.

**Comment by zeertzjq** [comment]
OK.

**Comment by clason** [comment]
Should probably also mention the requirement on `gh` in https://github.com/neovim/neovim/wiki/Merging-patches-from-upstream-Vim.

File: scripts/vim-patch.sh [hunk 1]
```diff
@@ -326,12 +326,14 @@ stage_patch() {
...
-  hub pull-request -m "$1"
+gh_pr() {
+  gh pr create --title "$1" --body "$2"
 }
 
 git_hub_pr() {
-  git hub pull new -m "$1"
+  local pr_message
+  pr_message="$(printf '%s\n\n%s\n' "$1" "$2")"
+  git hub pull new -m "${pr_message}"
```
Comment (teto) [inline scripts/vim-patch.sh:336]: it is still using hub.

**Comment by teto** [comment]
> Should probably also mention the requirement on `gh` in https://github.com/neovim/neovim/wiki/Merging-patches-from-upstream-Vim.

Yes, this section `Install hub or git-hub`  for instance.

File: scripts/vim-patch.sh (hunk 1 above)
Comment (jamessan) [reply scripts/vim-patch.sh]: No, [git-hub](https://github.com/sociomantic-tsunami/git-hub/) is a separate project.  Only hub should be replaced by gh.

Comment (teto) [reply scripts/vim-patch.sh:336]: While at it we could ditch git-hub too, depends on @janlazo that uses the script the most I guess.

**Review by jamessan — APPROVED** [review]

**Pull request merged**